src/main/java/org/beanpod/switchboard/controller/EncoderController.java

```diff
@@ -31,24 +33,36 @@
 @RequestMapping("/encoder")
 @RequiredArgsConstructor
 public class EncoderController {
+
   public static final String UNKNOWN_ERROR_MESSAGE = "Unknown error in EncoderController";
 
   private final EncoderDaoImpl encoderDao;
   private final DeviceDaoImpl deviceService;
   private final EncoderMapper encoderMapper;
   private final StreamMapper streamMapper;
   private final EncoderService encoderService;
+  private final MaintainDeviceStatus maintainDeviceStatus;
 
   @GetMapping
   public List<EncoderDto> retrieveAllEncoders() {
     List<EncoderEntity> encoderEntities = encoderDao.getEncoders();
+    maintainDeviceStatus.maintainStatusField(encoderEntities);
     return encoderMapper.toEncoderDtos(encoderEntities);
   }
 
   @GetMapping("/{serialNumber}")
   public ResponseEntity<EncoderDto> retrieveEncoder(@PathVariable @Valid String serialNumber) {
-    return encoderDao
-        .findEncoder(serialNumber)
+
+    // maintain status field and create a log if status changed
+    Optional<EncoderDto> encoder = encoderDao.findEncoder(serialNumber);
+    if (encoder.isPresent()) {
+      List<EncoderEntity> encodersListTemp = new LinkedList<>();
```

Comment:
can we pass just the object or is there a reason we are using a list

Reply:
I'm not a huge fan of the logging logic being used to maintain the status field. I think this could be ultimately refactored using aspect oriented programming. I'd suggest you look at the aspect classes done by @mohhef or consult with him as inspiration to refactor your code.

Reply:
@MrJCipherButtles I have extracted the logging logic to an Aspect class.

Reply:
@mohhef We could, but then maintainStatusField will be duplicated since the logic for retrieveAllEncoders and retrieveEncoder method is the same exact logic.